### ISSUE TITLE
Layout issue in the Label.

### ISSUE THREAD

**Issue by BDisp** [migueldeicaza/gui.cs#703]
In my opinion I think that the `LayoutSubviews ()` method should not be virtual because it is essential for the redisposition of all views and any overriden may forget to call the base. But for now I just put the Label to call the base method. What is your opinion?

![layout-issue](https://user-images.githubusercontent.com/13117724/85029038-c4d6fa80-b173-11ea-9d27-d2b50e52bb6d.gif)

**Comment by tig** [comment]
What will break if we do this? I think it sounds like a reasonable idea, but I don't know how much existing behavior depends on it.

**Comment by BDisp** [comment]
> 
> 
> What will break if we do this? I think it sounds like a reasonable idea, but I don't know how much existing behavior depends on it.

I did a search and just found it on `Label `and `Dialog `but this one he calls the `base `as it should be.

**Comment by tig** [comment]
It will break Dialog; The only way to remove it as a virtual and fix is to add another event to View (LayoutStarted). I just did this and will submit a PR for this issue. Ok?

**Comment by tig** [comment]
Actually, I spoke too soon. Adding a new event lets classes do their thing without relying on overriding, but if I change Layout to non-virtual I get an OOM error. This will require more investigation.

**Comment by tig** [comment]
```csharp
			LayoutStarted += (args) => {
				LayoutStartedHandler ();
			};

...


void LayoutStartedHandler ()
{
	int buttonsWidth = GetButtonsWidth ();

	int shiftLeft = Math.Max ((Bounds.Width - buttonsWidth) / 2 - 2, 0);
	for (int i = buttons.Count - 1; i >= 0; i--) {
		Button button = buttons [i];
		shiftLeft += button.Frame.Width + 1;
		button.X = Pos.AnchorEnd (shiftLeft);
		button.Y = Pos.AnchorEnd (1);
	}
}

```

**Comment by BDisp** [comment]
@tig realmente acho que é a melhor solução. Usar um evento e não override. Do it please. Thanks.

**Comment by BDisp** [comment]
@tig really think it is the best solution. Use an event and don't override. Do it please. Thanks.

**Comment by BDisp** [comment]
I'll close my PR too.

**Comment by BDisp** [comment]
@tig I already tested the code above and it seems working well.

**Comment by tig** [comment]
Did you change it away from virtual? If I do it i get an out of memory error when starting. Don't have time to look at it right now...

**Comment by BDisp** [comment]
Affirmative @tig. I don't get any errors after removing the virtual from `LayoutSubviews ()`. Perhaps it is another change that you have made that may be affecting this and not exactly because of this situation.

**Comment by BDisp** [comment]
@tig see it here https://github.com/BDisp/gui.cs/tree/massive-pr-refactored-text-rendering-alignment-word-wrap-hotkeys-all-now-work-with-unicode

**Comment by tig** [comment]
This will be fixed in #702
